### ISSUE TITLE
Error after upgrading 37 -> 38 (bare) due to expo-splash-screen dependency

### ISSUE THREAD

**Issue by hardcodet** [expo/expo#8995]
TL/DR: Manually adding the `expo-splash-screen` package fixed the issue for me.
EDIT: Tested the fix on Android only - apparently that's not sufficient on iOS (see answer below).

## 🐛 Bug Report

My app crashes on startup after upgrading my bare app:

```
error: Error: Unable to resolve module `expo-splash-screen` from `node_modules\expo\build\launch\SplashScreen.js`: expo-spla
sh-screen could not be found within the project.
```

Never used that package, looks like a dependency was introduced to it.

-- 

As a side note: that's yet another package I don't really use, and build times for my pretty basic expo app are through the roof already, which sort of makes the goal of bare workflows questionable: If every build cycle after a change in my single native module costs me 4 minutes, that's quite far from enjoyable...

**Comment by sunny-lirr** [comment]
Ran into the same issue. Should note that just adding expo-splash-screen without actually using it resulted in a build error on iOS, because I didn't have a `SplashScreen.storyboard` file in my app.

**Comment by shawnharvell** [comment]
also just hit this during upgrade of sdk37 -> sdk38

**Comment by zchwyng** [comment]
Same here.

**Comment by vitorbarros** [comment]
I got this issue after upgrading sdk38 to sdk39, any updates?

**Comment by peterchijioke** [comment]
how do i solve this error please.


 Error: No native splash screen registered for provided activity. Please configure your application's main Activity to call 'SplashScreen.show' (https://github.com/expo/expo/tree/master/packages/expo-splash-screen#-configure-android).]

**Comment by vitorbarros** [comment]
Hey guys any updates? I am using managed workflow and I am getting this error

**Comment by bbarthec** [comment]
@peterchijioke, seems you're reporting the different problem 🤔 Do you mind checking if your scenario isn't more like the once described here: https://github.com/expo/expo/issues/10263?

**Comment by bbarthec** [comment]
@vitorbarros, do you mind giving me more context on your problem?
Ideally, I wish you described me everything you did before encountering this error 🙏

**Comment by vitorbarros** [comment]
Hey, @bbarthec thanks for your reply.

I was working with expo 38 but there were some errors on iOS 14 that supposed to be fixed on expo 39, and it was.

I just ran `expo upgrade` I did not do anything else.

But after that, I got an error regarding the splash screen. `Error: No native splash screen registered for provided activity. Please configure your application's main Activity to call 'SplashScreen.show'`

the weirdest is that I am in the managed workflow.

Follow bellow my App.tsx render function

```
render() {
        if (!this.props.skipLoadingScreen && (!this.state.isLoadingComplete)) {
            return (
                <AppLoading
                    startAsync={this._loadStuffAsync}
                    onError={this._handleLoadingError}
                    onFinish={this._handleFinishLoading}
                />
            );
        }

        return (
            <MobxProvider>
                <View style={styles.container}>
                    <StatusBar barStyle="light-content" hidden={true} />
                    <AppRouter />
                </View>
            </MobxProvider>
        );
    }
```

and my app.json

```
{
  "expo": {
    "name": "...",
    "slug": "...",
    "privacy": "unlisted",
    "platforms": [
      "ios",
      "android"
    ],
    "version": "1.0.2",
    "orientation": "portrait",
    "icon": "./src/assets/appicons/appicon_evolution.png",
    "splash": {
    "resizeMode": "contain",
      "backgroundColor": "#344A5D"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "src/assets/images/*",
    ],
    "isDetached": false,
    "extra": {
      "softVersion": "1"
    },
    "ios": {
      "supportsTablet": false,
      "usesAppleSignIn": true,
      "icon": "./src/assets/appicons/appicon_evolution.png",
      "buildNumber": "1",
      "bundleIdentifier": "....",
      "config": {
        "usesNonExemptEncryption": false,
        "googleSignIn": {
          "reservedClientId": "..."
        }
      },
      "associatedDomains": [
        "...."
      ],
      "googleServicesFile": "./configs/evolution/GoogleService-Info.plist",
      "infoPlist": {
        "NSMicrophoneUsageDescription": "This allows us to store and process your daily audio journal recordings. You will be able to listen to them and delete them.",
        "NSPhotoLibraryUsageDescription": "We will use a selected photo or image from your library as an avatar for your profile."
      }
    },
    "android": {
      "config": {
        "googleSignIn": {
          "apiKey": "....",
          "certificateHash": "...."
        }
      },
      "package": "....",
      "googleServicesFile": "./configs/evolution/google-services.json",
      "versionCode": 1,
      "intentFilters": [
        {
          "action": "VIEW",
          "autoVerify": true,
          "data": [
            {
              "scheme": "....",
              "host": "...."
            }
          ],
          "category": [
            "BROWSABLE",
            "DEFAULT"
          ]
        }
      ]
    },
    "hooks": {
      "postPublish": [
        {
          "file": "sentry-expo/upload-sourcemaps",
          "config": {
            "organization": "....",
            "project": "....",
            "authToken": "..."
          }
        }
      ]
    },
    "packagerOpts": {
      "config": "metro.config.js",
      "sourceExts": [
        "js",
        "jsx",
        "ts",
        "tsx",
        "svg"
      ]
    }
  }
}

```

**Comment by bbarthec** [comment]
@vitorbarros, oh ok, so you're in managed workflow 😉
Indeed, managed workflow in SDK 39 has this problem (at least on iOS), but fortunately the fix is going to published soon. See https://github.com/expo/expo/issues/10263

**Comment by vitorbarros** [comment]
ok great!!!

**Comment by changLiuUNSW** [comment]
@hardcodet
I am not sure why but after I use ```yarn install``` instead ```npm install``` the issue is gone...
So for u, please do
```
rm -rf node_module
rm -rf package-lock.json
yarn install
cd ios
pod install
```

**Comment by bbarthec** [comment]
I'm closing this issue tentatively, but feel free to reopen or create a new one if this is still a problem for you 💪
